src/SingleArticle.jsx

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from "react"
+import { useParams } from "react-router-dom"
+import { getSingleArticle } from "../api"
+
+const SingleArticle = () => {
+  const { articleId } = useParams()
+  const [article, setArticle] = useState([])
+  const [loading, setLoading] = useState(true)
+
```

Comment:
A good practice to follow with State in react for reducing bugs is to _always keep a state's data type the same_ 

`article` is starting off as an array `[]` but then being changed to an object `{}` ( at least I hope your API call gives back an object for a single article ). This invites bugs as object methods will yield type errors 🐛